src/sema/contracts.rs

```diff
@@ -109,6 +110,20 @@ pub fn resolve(
     // Now we can resolve the initializers
     variables::resolve_initializers(&delayed.initializers, file_no, ns);
 
+    for (n, _) in contracts {
+        let contract = &ns.contracts[*n];
+        if ns.target.is_substrate()
+            && contract.is_concrete()
+            && !contract.has_callable_function
+            && !ns.diagnostics.any_errors()
+        {
+            ns.diagnostics.push(ast::Diagnostic::error(
+            contract.loc,
+            "contracts without public storage or functions are not allowed on Substrate, considering declaring it abstract".into(),
```

Comment:
There is no test case for this error

Reply:
Sorry forgot to make it draft, I'm working on adding tests for this as well